### PR TITLE
Issue/#789

### LINKED ISSUE #789: 'Global' configs for plugins should be conditionally applied
For example, adding the Page Level Progress extension should add the global configs to the course.  Removing it should remove the settings/values.

### PR DESCRIPTION
**Pull request by darylhedley**
fixes #789

### CODE REVIEW

File: plugins/content/config/index.js [hunk 1]
```diff
@@ -121,7 +122,34 @@ ConfigContent.prototype.retrieve = function (search, options, next) {
...
+
+        var uniqueComponentList = _.map(_.uniq(components, false, function(component) {
+          return component._component;
+        }), function(component) {
+          // Adding an _ here is necessary because of the way the attributes are set
+          // onto the _globals object on the schemas
+          return {_componentType: component._componentType, _component: "_" + component._component};
+        })
+
+        var configModel = records[0].toObject()
```
Comment (dennis-learningpool) [inline plugins/content/config/index.js]: missing `;` at the end

File: plugins/content/config/index.js [hunk 2]
```diff
@@ -121,7 +122,34 @@ ConfigContent.prototype.retrieve = function (search, options, next) {
...
+    }
+    // When passing down the config model we should 
+    // find out how many and what components are being used
+    // Why you might ask - to sort out _globals and when compiling
+    // this should act as a filter
+    app.contentmanager.getContentPlugin('component', function (err, componentPlugin) {
+      var courseId = records[0]._courseId;
+      componentPlugin.retrieve({_courseId: courseId}, function(err, components) {
+
+        var uniqueComponentList = _.map(_.uniq(components, false, function(component) {
```
Comment (dennis-learningpool) [inline plugins/content/config/index.js]: should this use [async.map](https://github.com/caolan/async#map)?

**Comment by dennis-learningpool** [comment]
+1 :)

**Comment by brian-learningpool** [comment]
+1

**Pull request merged**